collections/_article/godotcon-2025-xr-android-recap.md

```diff
@@ -0,0 +1,135 @@
+
+The Android platform has numerous APIs as well as a rich ecosystem of third-party libraries with wide and diverse functionality. Godot has long provided an [Android plugin system](https://docs.godotengine.org/en/stable/tutorials/platform/android/android_plugin.html) which enables developers to create a Godot interface in order to access and use Android APIs or third-party libraries in their projects.  
+Writing an Android plugin however requires knowledge of Java or Kotlin code, which most Godot developers do not have. As such there are many Android APIs and third-party libraries that don’t have a Godot plugin that Godot developers can interface with. In fact, this is one of the main reasons that developers cite for not being able to switch to Godot from other game engines.
+
+To address this, we’ve introduced a couple of tools in Godot 4.4 to simplify the process for developers to **access Android APIs and third-party libraries**.
+
+## JavaClassWrapper and AndroidRuntime plugin
+
+**JavaClassWrapper** is a Godot singleton which allows for creating instances of Java and Kotlin classes and calling methods on them using only GDScript, C\#, or GDExtension.
+
+**AndroidRuntime plugin** is a built-in Godot Android plugin that provides access to various Android lifecycle and runtime objects.
+
+Coupling the two together allows developers to access and use Android APIs without switching away from GDScript, or using any tools aside from Godot itself.  
+This is **huge** for the adoption of Godot for Android development, as it allows developers to quickly integrate Android functionality without having to make a Java or Kotlin plugin\!
+
+For example, this code snippet is all that developers need to access and use the [Android Vibrator system service](https://developer.android.com/reference/android/os/Vibrator):
+
+```gdscript
+    # Retrieve the AndroidRuntime singleton
+    var android_runtime = Engine.get_singleton("AndroidRuntime")
+    if android_runtime:
+
+        # Retrieve the Android Vibrator system service and check if the device supports it
+        var vibrator_service = android_runtime.getApplicationContext().getSystemService("vibrator")
+        if vibrator_service and vibrator_service.hasVibrator():
+
+            # Configure and run a VibrationEffect 
+            var VibrationEffect = JavaClassWrapper.wrap("android.os.VibrationEffect")
+            var effect = VibrationEffect.createOneShot(500, VibrationEffect.DEFAULT_AMPLITUDE)
+            vibrator_service.vibrate(effect)
+```
+
+To learn more about these new capabilities, please check out the [documentation](https://docs.godotengine.org/en/latest/tutorials/platform/android/javaclasswrapper_and_androidruntimeplugin.html).
+
+# And more\!
+
+There’s plenty more we covered in our GodotCon talk\! For the full content, please check out the  presentation slides at [https://bit.ly/godotcon-boston-2025-xr-android](https://bit.ly/godotcon-boston-2025-xr-android).  
```

Comment:
I would add a reminder for the video again here, and also maybe thank the people leading this effort.